src/query/atomWithQuery.ts

```diff
@@ -95,10 +95,12 @@ export function atomWithQuery<
         defaultedOptions.staleTime = 1000
       }
       const observer = new QueryObserver(queryClient, defaultedOptions)
-      observer
-        .fetchOptimistic(defaultedOptions)
-        .then(listener)
-        .catch((error) => listener({ error }))
+      if (!defaultedOptions.initialData) {
```

Comment:
More precisely, you want to check if `getInitialData()` returns non-falsy value. Like, `options.initialData = () => null`. We may drop the support of it. Just needs to make it consistent.

Reply:
Sure, that is a good catch. Do you think this would be sufficient?
```suggestion
      if (!getInitialData()) {
```

Reply:
I think we should call `getInitialData()` just once in our code.

Reply:
yep, agreed!